common/sagas/schedule/paramsValidity.ts

```diff
@@ -0,0 +1,154 @@
+import { SagaIterator, delay } from 'redux-saga';
+import { select, fork, call, take, apply, put } from 'redux-saga/effects';
+import { getOffline, getNodeLib } from 'selectors/config';
+import {
+  ICurrentSchedulingToggle,
+  ICurrentWindowSize,
+  ICurrentWindowStart,
+  ICurrentScheduleType,
+  ICurrentScheduleTimezone,
+  ICurrentScheduleTimestamp,
+  ICurrentTimeBounty,
+  getSchedulingToggle,
+  getScheduleTimestamp,
+  getScheduleTimezone,
+  getScheduleType,
+  getWindowStart,
+  getWindowSize,
+  getTimeBounty,
+  getScheduleGasPrice,
+  getScheduleGasLimit,
+  getScheduleDeposit
+} from 'selectors/schedule/fields';
+import {
+  TypeKeys,
+  SetScheduleParamsValidityAction,
+  setScheduleParamsValidity
+} from 'actions/schedule';
+import { TypeKeys as TransactionTypeKeys } from 'actions/transaction';
+import { getData } from 'selectors/transaction/fields';
+import { getWalletInst } from 'selectors/wallet';
+import {
+  EAC_SCHEDULING_CONFIG,
+  calcEACEndowment,
+  getValidateRequestParamsData,
+  EAC_ADDRESSES,
+  parseSchedulingParametersValidity
+} from 'libs/scheduling';
+import { gasPriceToBase } from 'libs/units';
+import BN from 'bn.js';
+import { bufferToHex } from 'ethereumjs-util';
+import RequestFactory from 'libs/scheduling/contracts/RequestFactory';
+import { windowSizeBlockToMin, calculateWindowStart } from 'selectors/schedule/helpers';
+import { getCurrentTo, getCurrentValue } from 'selectors/transaction/current';
+
+export function* shouldValidateParams(): SagaIterator {
+  while (true) {
+    yield take([
+      TransactionTypeKeys.TO_FIELD_SET,
+      TransactionTypeKeys.DATA_FIELD_SET,
+      TypeKeys.CURRENT_TIME_BOUNTY_SET,
+      TypeKeys.WINDOW_SIZE_FIELD_SET,
+      TypeKeys.WINDOW_START_FIELD_SET,
+      TypeKeys.SCHEDULE_TIMESTAMP_FIELD_SET,
+      TypeKeys.TIME_BOUNTY_FIELD_SET,
+      TypeKeys.SCHEDULE_TYPE_SET,
+      TypeKeys.SCHEDULING_TOGGLE_SET,
+      TypeKeys.SCHEDULE_TIMEZONE_SET
+    ]);
+
+    yield call(delay, 250);
+
+    const isOffline: boolean = yield select(getOffline);
+    const schedulingToggle: ICurrentSchedulingToggle = yield select(getSchedulingToggle);
+    const scheduling = Boolean(schedulingToggle && schedulingToggle.value);
+
+    if (isOffline || !scheduling) {
+      continue;
+    }
+
+    yield call(checkSchedulingParametersValidity);
+  }
+}
+
+function* checkSchedulingParametersValidity() {
```

Comment:
The majority of this function can instead be computed in a single selector + getValidateRequestParamsData can be called from a library function. This would make testing much easier since simple functions require a lot less effort to test vs sagas. LN #132 + #144 is what is only really needed to be inside the saga, everything else can be computed outside and called.

Reply:
The majority of the function has been extracted to its own separate selector.